tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/es/UserTaskZeebeRecordProcessorElasticSearch.java

```diff
@@ -130,4 +135,72 @@ private UpdateRequest getVariableQuery(final TaskVariableEntity variable)
+    Optional.ofNullable(taskEntity.getFlowNodeBpmnId())
+        .ifPresent(userTaskListViewEntity::setTaskId);
+    Optional.ofNullable(taskEntity.getFlowNodeBpmnId())
+        .ifPresent(userTaskListViewEntity::setFlowNodeBpmnId);
+    Optional.of(taskEntity.getKey()).ifPresent(userTaskListViewEntity::setKey);
+    Optional.of(taskEntity.getPartitionId()).ifPresent(userTaskListViewEntity::setPartitionId);
+    Optional.ofNullable(taskEntity.getCompletionTime())
+        .map(Object::toString)
+        .ifPresent(userTaskListViewEntity::setCompletionTime);
+    Optional.ofNullable(taskEntity.getAssignee()).ifPresent(userTaskListViewEntity::setAssignee);
+    Optional.ofNullable(taskEntity.getCreationTime())
+        .map(Object::toString)
+        .ifPresent(userTaskListViewEntity::setCreationTime);
+    Optional.ofNullable(taskEntity.getProcessDefinitionVersion())
+        .ifPresent(userTaskListViewEntity::setProcessDefinitionVersion);
+    Optional.ofNullable(taskEntity.getPriority()).ifPresent(userTaskListViewEntity::setPriority);
+    Optional.ofNullable(taskEntity.getCandidateGroups())
+        .ifPresent(userTaskListViewEntity::setCandidateGroups);
+    Optional.ofNullable(taskEntity.getCandidateUsers())
+        .ifPresent(userTaskListViewEntity::setCandidateUsers);
+    Optional.ofNullable(taskEntity.getBpmnProcessId())
+        .ifPresent(userTaskListViewEntity::setBpmnProcessId);
+    Optional.ofNullable(taskEntity.getProcessDefinitionId())
+        .ifPresent(userTaskListViewEntity::setProcessDefinitionId);
+    Optional.ofNullable(taskEntity.getTenantId()).ifPresent(userTaskListViewEntity::setTenantId);
+    Optional.ofNullable(taskEntity.getExternalFormReference())
+        .ifPresent(userTaskListViewEntity::setExternalFormReference);
+    Optional.ofNullable(taskEntity.getCustomHeaders())
+        .ifPresent(userTaskListViewEntity::setCustomHeaders);
+    Optional.ofNullable(taskEntity.getFormKey()).ifPresent(userTaskListViewEntity::setFormKey);
+    Optional.ofNullable(taskEntity.getState()).ifPresent(userTaskListViewEntity::setState);
+
+    userTaskListViewEntity.setJoin(new ListViewJoinRelation());
+    // Set Join Field for the parent
+    userTaskListViewEntity.getJoin().setName("task");
+    userTaskListViewEntity
+        .getJoin()
+        .setParent(Long.valueOf(String.valueOf(taskEntity.getProcessInstanceId())));
+
+    return userTaskListViewEntity;
+  }
+
+  private UpdateRequest persistUserTaskToListView(
+      final UserTaskListViewEntity userTaskListViewEntity) throws PersistenceException {
+    try {
+      final Map<String, Object> jsonMap =
+          objectMapper.readValue(
+              objectMapper.writeValueAsString(userTaskListViewEntity), HashMap.class);
+      return new UpdateRequest()
+          .index(tasklistListViewTemplate.getFullQualifiedName())
+          .id(userTaskListViewEntity.getId())
+          .upsert(objectMapper.writeValueAsString(userTaskListViewEntity), XContentType.JSON)
```

Comment:
in update operations, we don't update all the fields, it depends on the record intent. checkout https://github.com/camunda/camunda/blob/41c8b7dd6b34c5b4678800fcaa74a4c7ef1e1f06/tasklist/importer-850/src/main/java/io/camunda/tasklist/zeebeimport/v850/processors/common/UserTaskRecordToTaskEntityMapper.java#L151

Reply:
True, but as I am creating object, this will update all fields (as tested already)

Similar as it is used on `private UpdateRequest getTaskQuery`

So at this case I dont see a case that it wont work this UpdateRequest

Reply:
I see,

Is there a reason this class is named as [UserTaskZeebeRecordProcessorElasticSearch.java](https://github.com/camunda/camunda/pull/21876/files/79d6cd32e9a5b9600d2d64b1d74fa429fd4cd90b#diff-965b4340bebda53d032628731dbf84098eb3c5b498e8b0e235c675e8f8c2a1c7)
And just perform updates?

This induces me to think that it would deal only with UserTask, but it looks Job is dealing with User Task & Job Worker.

Reply:
I update this with the method getUpdateFieldsListViewMap (As the entity and template is different, I did a similar logic) on this commit: https://github.com/camunda/camunda/pull/21876/commits/bde3ae5579c094bb580d5b66cf60e0351d63b154